JDBC instrumentation NullPointerException for TRUNCATE

### Describe the bug

The crash is a NullPointerException inside agent code (AgentContextWrapper.get), which should generally be null-safe.

Failing when running TRUNCATE /DDL flow in Trino, applicationContext is null.
This error is crashing Trino.

Stacktrace:
`com.bearingpoint.mpe.engine.TaskExecutionException: Error occurred for task of type Case (at ./configuration/engine/params/mpe/mpeBuildDB.xml:425):Error in case-when construct: Error occurred for task of type If (at ./configuration/engine/params/mpe/mpeBuildDB.xml:333):Error in if construction: java.sql.SQLException: Query failed (#20260227_140924_06021_5tuc3): Cannot invoke "io.opentelemetry.context.Context.get(io.opentelemetry.context.ContextKey)" because "this.applicationContext" is null 	at com.bearingpoint.mpe.engine.BaseTask.error(BaseTask.java:453) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.Case.onExecution(Case.java:88) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.onExecution(BaseTask.java:426) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.exec(BaseTask.java:342) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.Sequence.onExecution(Sequence.java:30) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.onExecution(BaseTask.java:426) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.exec(BaseTask.java:342) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.Document.onExecution(Document.java:106) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.onExecution(BaseTask.java:426) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.exec(BaseTask.java:342) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.davinci.database.builddb.Build.runBuildProcess(Build.java:606) [abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.davinci.database.builddb.Build.execute(Build.java:575) [abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.davinci.database.builddb.Build.main(Build.java:691) [abacus-mpe-engine-new.jar:?] Caused by: com.bearingpoint.mpe.engine.TaskExecutionException: Error occurred for task of type If (at ./configuration/engine/params/mpe/mpeBuildDB.xml:333):Error in if construction: java.sql.SQLException: Query failed (#20260227_140924_06021_5tuc3): Cannot invoke "io.opentelemetry.context.Context.get(io.opentelemetry.context.ContextKey)" because "this.applicationContext" is null 	at com.bearingpoint.mpe.engine.BaseTask.error(BaseTask.java:453) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.If.onExecution(If.java:71) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.onExecution(BaseTask.java:426) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.exec(BaseTask.java:342) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.Sequence.onExecution(Sequence.java:30) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.onExecution(BaseTask.java:426) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.exec(BaseTask.java:342) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.Execute.onExecution(Execute.java:54) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.onExecution(BaseTask.java:426) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.exec(BaseTask.java:342) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.Sequence.onExecution(Sequence.java:30) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.onExecution(BaseTask.java:426) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.exec(BaseTask.java:342) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.Case.onExecution(Case.java:78) ~[abacus-mpe-engine-new.jar:?] 	... 11 more Caused by: com.bearingpoint.mpe.engine.TaskExecutionException: java.sql.SQLException: Query failed (#20260227_140924_06021_5tuc3): Cannot invoke "io.opentelemetry.context.Context.get(io.opentelemetry.context.ContextKey)" because "this.applicationContext" is null 	at com.bearingpoint.mpe.mpengine.NativeClass.exec(NativeClass.java:241) ~[abacus-mpe-engine-common.jar:?] 	at com.bearingpoint.mpe.engine.NativeDeclaration.onExecution(NativeDeclaration.java:67) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.onExecution(BaseTask.java:426) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.exec(BaseTask.java:342) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.Execute.onExecution(Execute.java:54) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.onExecution(BaseTask.java:426) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.exec(BaseTask.java:342) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.Sequence.onExecution(Sequence.java:30) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.If.onExecution(If.java:60) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.onExecution(BaseTask.java:426) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.exec(BaseTask.java:342) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.Sequence.onExecution(Sequence.java:30) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.onExecution(BaseTask.java:426) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.exec(BaseTask.java:342) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.Execute.onExecution(Execute.java:54) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.onExecution(BaseTask.java:426) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.exec(BaseTask.java:342) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.Sequence.onExecution(Sequence.java:30) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.onExecution(BaseTask.java:426) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.BaseTask.exec(BaseTask.java:342) ~[abacus-mpe-engine-new.jar:?] 	at com.bearingpoint.mpe.engine.Case.onExecution(Case.java:78) ~[abacus-mpe-engine-new.jar:?] 	... 11 more Caused by: java.sql.SQLException: Query failed (#20260227_140924_06021_5tuc3): Cannot invoke "io.opentelemetry.context.Context.get(io.opentelemetry.context.ContextKey)" because "this.applicationContext" is null 	at io.trino.jdbc.ResultUtils.resultsException(ResultUtils.java:33) ~[trino-jdbc-476.jar:476] 	at io.trino.jdbc.AsyncResultIterator.lambda$new$1(AsyncResultIterator.java:94) ~[trino-jdbc-476.jar:476] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?] 	at java.base/java.lang.Thread.run(Thread.java:840) ~[?:?] Caused by: io.trino.jdbc.$internal.client.FailureException: Cannot invoke "io.opentelemetry.context.Context.get(io.opentelemetry.context.ContextKey)" because "this.applicationContext" is null 	at io.opentelemetry.javaagent.instrumentation.opentelemetryapi.context.AgentContextWrapper.get(AgentContextWrapper.java:82) ~[?:?] 	at io.opentelemetry.api.trace.Span.fromContext(Span.java:49) ~[?:?] 	at io.opentelemetry.instrumentation.api.instrumenter.Instrumenter.doEnd(Instrumenter.java:243) ~[?:?] 	at io.opentelemetry.instrumentation.api.instrumenter.Instrumenter.end(Instrumenter.java:151) ~[?:?] 	at io.opentelemetry.instrumentation.jdbc.internal.OpenTelemetryStatement.wrapCall(OpenTelemetryStatement.java:384) ~[?:?] 	at io.opentelemetry.instrumentation.jdbc.internal.OpenTelemetryStatement.wrapCall(OpenTelemetryStatement.java:365) ~[?:?] 	at io.opentelemetry.instrumentation.jdbc.internal.OpenTelemetryStatement.execute(OpenTelemetryStatement.java:95) ~[?:?] 	at io.trino.plugin.jdbc.BaseJdbcClient.execute(BaseJdbcClient.java:1628) ~[?:?] 	at io.trino.plugin.jdbc.BaseJdbcClient.execute(BaseJdbcClient.java:1615) ~[?:?] 	at io.trino.plugin.jdbc.BaseJdbcClient.truncateTable(BaseJdbcClient.java:1793) ~[?:?] 	at io.trino.plugin.jdbc.ForwardingJdbcClient.truncateTable(ForwardingJdbcClient.java:512) ~[?:?] 	at io.trino.plugin.jdbc.jmx.StatisticsAwareJdbcClient.lambda$truncateTable$0(StatisticsAwareJdbcClient.java:533) ~[?:?] 	at io.trino.plugin.jdbc.jmx.JdbcApiStats.wrap(JdbcApiStats.java:46) ~[?:?] 	at io.trino.plugin.jdbc.jmx.StatisticsAwareJdbcClient.truncateTable(StatisticsAwareJdbcClient.java:533) ~[?:?] 	at io.trino.plugin.jdbc.RetryingJdbcClient.truncateTable(RetryingJdbcClient.java:532) ~[?:?] 	at io.trino.plugin.jdbc.CachingJdbcClient.truncateTable(CachingJdbcClient.java:683) ~[?:?] 	at io.trino.plugin.jdbc.CachingJdbcClient.truncateTable(CachingJdbcClient.java:683) ~[?:?] 	at io.trino.plugin.jdbc.DefaultJdbcMetadata.truncateTable(DefaultJdbcMetadata.java:1377) ~[?:?] 	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorMetadata.truncateTable(ClassLoaderSafeConnectorMetadata.java:460) ~[?:?] 	at io.trino.tracing.TracingConnectorMetadata.truncateTable(TracingConnectorMetadata.java:397) ~[?:?] 	at io.trino.metadata.MetadataManager.truncateTable(MetadataManager.java:1048) ~[?:?] 	at io.trino.tracing.TracingMetadata.truncateTable(TracingMetadata.java:556) ~[?:?] 	at io.trino.execution.TruncateTableTask.execute(TruncateTableTask.java:78) ~[?:?] 	at io.trino.execution.TruncateTableTask.execute(TruncateTableTask.java:36) ~[?:?] 	at io.trino.execution.DataDefinitionExecution.start(DataDefinitionExecution.java:152) ~[?:?] 	at io.trino.execution.SqlQueryManager.createQuery(SqlQueryManager.java:272) ~[?:?] 	at io.trino.dispatcher.LocalDispatchQuery.startExecution(LocalDispatchQuery.java:150) ~[?:?] 	at io.trino.dispatcher.LocalDispatchQuery.lambda$waitForMinimumWorkers$1(LocalDispatchQuery.java:134) ~[?:?] 	at io.airlift.concurrent.MoreFutures.lambda$addSuccessCallback$12(MoreFutures.java:570) ~[?:?] 	at io.airlift.concurrent.MoreFutures$3.onSuccess(MoreFutures.java:545) ~[?:?] 	at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1132) ~[guava-33.2.1-jre.jar:?] 	at io.trino.$gen.Trino_476____20260227_130022_2.run(Unknown Source) ~[?:?] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1095) ~[?:?] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:619) ~[?:?] 	at java.base/java.lang.Thread.run(Thread.java:1447) ~[?:?]`

### Steps to reproduce

Run Trino 476 image with included `opentelemetry-javaagent.jar` 2.25.0 and traces through OTEL Agent on K8s cluster (GKE).

Trino helm chart values (cleaned up version - not sure it will deploy):
```
image:
  repository: trinodb/trino
  tag: "476"

server:
  workers: 1
  node:
    environment: "test"
  log:
    level: "INFO"
  config:
    query:
      maxMemory: "1GB"

additionalConfigProperties:
  - log.console-format=JSON
  - log.format=JSON

env:
  - name: OTEL_EXPORTER_OTLP_ENDPOINT
    value: "http://otel-collector-opentelemetry-collector.otel.svc.cluster.local:4317"
  - name: OTEL_EXPORTER_OTLP_PROTOCOL
    value: "grpc"
  - name: OTEL_JMX_TARGET_SYSTEM
    value: "jvm,jetty"
  - name: OTEL_METRICS_EXPORTER
    value: "otlp"
  - name: OTEL_TRACES_EXPORTER
    value: "otlp"
  - name: OTEL_LOGS_EXPORTER
    value: "none"
  - name: TRINO_CLUSTER
    valueFrom:
      fieldRef:
        fieldPath: metadata.labels['app.kubernetes.io/instance']
  - name: POD_NAME
    valueFrom:
      fieldRef:
        fieldPath: metadata.name
  - name: POD_NAMESPACE
    valueFrom:
      fieldRef:
        fieldPath: metadata.namespace
  - name: NODE_NAME
    valueFrom:
      fieldRef:
        fieldPath: spec.nodeName
  - name: TRINO_ROLE
    valueFrom:
      fieldRef:
        fieldPath: metadata.labels['trino_role']
  # Make GMP prometheus_target.job identify Trino cluster and role.
  - name: OTEL_SERVICE_NAME
    value: "$(TRINO_CLUSTER)-$(TRINO_ROLE)"
  - name: OTEL_RESOURCE_ATTRIBUTES
    value: "trino_cluster=$(TRINO_CLUSTER),trino_role=$(TRINO_ROLE),node_name=$(NODE_NAME),k8s.namespace.name=$(POD_NAMESPACE),k8s.pod.name=$(POD_NAME),k8s.node.name=$(NODE_NAME),service.instance.id=$(POD_NAME)@$(NODE_NAME)"

initContainers:
  coordinator:
    - name: otel-javaagent
      # Fetch from GitHub; requires outbound egress.
      image: curlimages/curl:8.12.0
      imagePullPolicy: IfNotPresent
      command:
        - sh
        - -c
        - >-
          set -e;
          echo "Init container start";
          sleep 5;
          echo "Preparing assets...";
          curl -fsSL -o /otel/opentelemetry-javaagent.jar https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/download/v2.25.0/opentelemetry-javaagent.jar;
          echo "Init container completed successfully"
      volumeMounts:
        - name: otel-javaagent
          mountPath: /otel
  worker:
    - name: otel-javaagent
      # Fetch from GitHub; requires outbound egress.
      image: curlimages/curl:8.12.0
      imagePullPolicy: IfNotPresent
      command:
        - sh
        - -c
        - >-
          set -e;
          echo "Init container start";
          sleep 5;
          echo "Preparing assets...";
          curl -fsSL -o /otel/opentelemetry-javaagent.jar https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/download/v2.25.0/opentelemetry-javaagent.jar;
          echo "Init container completed successfully"
      volumeMounts:
        - name: otel-javaagent
          mountPath: /otel

service:
  type: LoadBalancer

serviceAccount:
  create: false
  name: "trino-ksa"

coordinator:
  resources:
    requests:
      cpu: "500m"
      memory: "1Gi"
    limits:
      cpu: "1"
      memory: "2Gi"
  jvm:
    maxHeapSize: "1G"
  additionalJVMConfig:
    - "-javaagent:/otel/opentelemetry-javaagent.jar"
  config:
    nodeScheduler:
      includeCoordinator: true
    query:
      maxMemoryPerNode: "512MB"
    memory:
      heapHeadroomPerNode: "256MB"
  labels:
    app: "trino"
    trino_role: "coordinator"
  # Shared volume for the OpenTelemetry Java agent (jar added in a later step).
  additionalVolumes:
    - name: otel-javaagent
      emptyDir: {}
  additionalVolumeMounts:
    - name: otel-javaagent
      mountPath: /otel
      readOnly: true

worker:
  replicas: 1
  resources:
    requests:
      cpu: "500m"
      memory: "1Gi"
    limits:
      cpu: "1"
      memory: "2Gi"
  jvm:
    maxHeapSize: "1G"
  additionalJVMConfig:
    - "-javaagent:/otel/opentelemetry-javaagent.jar"
  config:
    query:
      maxMemoryPerNode: "512MB"
    memory:
      heapHeadroomPerNode: "256MB"
  labels:
    app: "trino"
    trino_role: "worker"
  # Shared volume for the OpenTelemetry Java agent (jar added in a later step).
  additionalVolumes:
    - name: otel-javaagent
      emptyDir: {}
  additionalVolumeMounts:
    - name: otel-javaagent
      mountPath: /otel
      readOnly: true

catalogs:
  tpch: |
    connector.name=tpch
```

### Expected behavior

No errors in OTEL Agent when truncating a table with Trino SQL DDL. No Trino crash.

### Actual behavior

Trino app is crashing and the Pod is restarting. CrashLoopback issue.

### Javaagent or library instrumentation version

2.25.0

### Environment

**JDK**: 21 (Trino 476 image)
**OS**: GKE k8s cluster 1.33.5-gke.2392000


### Additional context

_No response_

### Tip

<sub>[React](https://github.blog/news-insights/product-news/add-reactions-to-pull-requests-issues-and-comments/) with 👍 to help prioritize this issue. Please use comments to provide useful context, avoiding `+1` or `me too`, to help us triage it. Learn more [here](https://opentelemetry.io/community/end-user/issue-participation/).</sub>

--- a/instrumentation/opentelemetry-api/opentelemetry-api-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/context/AgentContextStorage.java
+++ b/instrumentation/opentelemetry-api/opentelemetry-api-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/context/AgentContextStorage.java
@@ -137,7 +137,7 @@ public class AgentContextStorage
 
   public static application.io.opentelemetry.context.Context toApplicationContext(
       Context agentContext) {
-    return new AgentContextWrapper(agentContext);
+    return AgentContextWrapper.getApplicationContext(agentContext);
   }
 
   public static application.io.opentelemetry.context.Context newContextWrapper(

--- a/instrumentation/opentelemetry-api/opentelemetry-api-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/context/AgentContextWrapper.java
+++ b/instrumentation/opentelemetry-api/opentelemetry-api-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/context/AgentContextWrapper.java
@@ -46,10 +46,6 @@ final class AgentContextWrapper implements application.io.opentelemetry.context.
   final Context agentContext;
   final application.io.opentelemetry.context.Context applicationContext;
 
-  AgentContextWrapper(Context agentContext) {
-    this(agentContext, agentContext.get(AgentContextStorage.APPLICATION_CONTEXT));
-  }
-
   AgentContextWrapper(
       Context agentContext, application.io.opentelemetry.context.Context applicationContext) {
     if (applicationContext instanceof AgentContextWrapper) {
@@ -59,15 +55,31 @@ final class AgentContextWrapper implements application.io.opentelemetry.context.
     this.applicationContext = applicationContext;
   }
 
+  public static application.io.opentelemetry.context.Context getApplicationContext(
+      Context agentContext) {
+    application.io.opentelemetry.context.Context rootApplicationContext =
+        application.io.opentelemetry.context.Context.root();
+    if (agentContext == Context.root()) {
+      return rootApplicationContext;
+    }
+
+    application.io.opentelemetry.context.Context applicationContext =
+        agentContext.get(AgentContextStorage.APPLICATION_CONTEXT);
+    // application context can be null when the agent context is converted to application context
+    // from instrumented code e.g. instrumentation uses
+    // AgentContextStorage.toApplicationContext(agentContext) to get application context but agent
+    // context isn't really associated with application context
+    if (applicationContext == null) {
+      applicationContext = ((AgentContextWrapper) rootApplicationContext).applicationContext;
+    }
+    return new AgentContextWrapper(agentContext, applicationContext);
+  }
+
   Context toAgentContext() {
     if (agentContext.get(AgentContextStorage.APPLICATION_CONTEXT) == applicationContext) {
       return agentContext;
     }
     return agentContext.with(AgentContextStorage.APPLICATION_CONTEXT, applicationContext);
-  }
-
-  public Context getAgentContext() {
-    return agentContext;
   }
 
   @Override

--- a/instrumentation/opentelemetry-extension-kotlin-1.0/javaagent/src/test/kotlin/io/opentelemetry/javaagent/instrumentation/extensionkotlin/v1_0/ContextExtensionInstrumentationTest.kt
+++ b/instrumentation/opentelemetry-extension-kotlin-1.0/javaagent/src/test/kotlin/io/opentelemetry/javaagent/instrumentation/extensionkotlin/v1_0/ContextExtensionInstrumentationTest.kt
@@ -9,6 +9,7 @@ import io.opentelemetry.context.Context
 import io.opentelemetry.context.ContextKey
 import io.opentelemetry.extension.kotlin.asContextElement
 import io.opentelemetry.extension.kotlin.getOpenTelemetryContext
+import kotlinx.coroutines.runBlocking
 import org.assertj.core.api.Assertions.assertThat
 import org.junit.jupiter.api.Test
 
@@ -27,5 +28,13 @@ class ContextExtensionInstrumentationTest {
     // instrumentation does not preserve context identity due to conversion between application and
     // agent context
     assertThat(context1).isNotSameAs(context2)
+  }
+
+  @Test
+  fun `no context`() {
+    runBlocking {
+      assertThat(coroutineContext.getOpenTelemetryContext()).isEqualTo(Context.root())
+      assertThat(coroutineContext.getOpenTelemetryContext().get(animalKey)).isNull()
+    }
   }
 }

